invenio_records_marc21/services/record/metadata.py

```diff
@@ -72,45 +77,47 @@ def load(self, xml: etree):
     def _to_xml_tree_from_string(self, xml: str):
         """Xml string to internal representation method."""
         tree = etree.parse(StringIO(xml))
-        self._to_xml_tree(tree)
         return tree
 
-    def _to_xml_tree(self, xml: etree):
-        """Xml to internal representation method."""
-        for element in xml.iter():
-            if "leader" in element.tag:
-                self.leader = LeaderField(data=element.text)
-            elif "datafield" in element.tag:
-                self.datafields.append(
-                    DataField(**element.attrib, subfields=element.getchildren())
-                )
-            elif "controlfield" in element.tag:
-                self.controlfields.append(ControlField(**element.attrib))
-
     def _to_string(self, tagsep: str = linesep, indent: int = 4) -> str:
         """Get a pretty-printed XML string of the record."""
-        self._xml = "<?xml version='1.0' ?>"
-        self._xml += '<record xmlns="http://www.loc.gov/MARC21/slim" xsi:schemaLocation="http://www.loc.gov/MARC21/slim schema.xsd" type="Bibliographic" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">'
-        self._xml += tagsep
-        if self.leader:
-            self._xml += self.leader.to_xml_tag(tagsep, indent)
-        for controlfield in self.controlfields:
-            self._xml += controlfield.to_xml_tag(tagsep, indent)
-        for datafield in self.datafields:
-            self._xml += datafield.to_xml_tag(tagsep, indent)
-        self._xml += "</record>"
-
-    def contains(self, ref_df: DataField, ref_sf: SubField) -> bool:
-        """Return True if record contains reference datafield, which contains reference subfield."""
-        for df in self.datafields:
-            if (
-                df.tag == ref_df.tag and df.ind1 == ref_df.ind1
-            ) and df.ind2 == ref_df.ind2:
-                for sf in df.subfields:
-                    if sf.code == ref_sf.code and sf.value == ref_sf.value:
-                        return True
+        self._xml = etree.tostring(self._etree, pretty_print=True).decode("UTF-8")
+
+    def contains(self, ref_df: dict, ref_sf: dict):
```

Comment:
what means ref_df and ref_sf?

Reply:
@param ref_df dict: datafield element specific information, containing keys [tag,ind1,ind2]
 @param ref_sf dict: subfield element specific information, containing keys [code,value]